pilot/pkg/status/report.go

```diff
@@ -48,22 +49,22 @@ func ResourceFromString(s string) *Resource {
 			Version:  pieces[1],
 			Resource: pieces[2],
 		},
-		Namespace:       pieces[3],
-		Name:            pieces[4],
-		ResourceVersion: pieces[5],
+		Namespace:  pieces[3],
+		Name:       pieces[4],
+		Generation: pieces[5],
 	}
 }
 
 // TODO: maybe replace with a kubernetes resource identifier, if that's a thing
 type Resource struct {
 	schema.GroupVersionResource
-	Namespace       string
-	Name            string
-	ResourceVersion string
+	Namespace  string
+	Name       string
+	Generation string
 }
 
 func (r Resource) String() string {
-	return strings.Join([]string{r.Group, r.Version, r.Resource, r.Namespace, r.Name, r.ResourceVersion}, "/")
+	return strings.Join([]string{r.Group, r.Version, r.Resource, r.Namespace, r.Name, r.Generation}, "/")
```

Comment:
Will this work for upgrades? when old istiod reports resource version and new one generation?

Reply:
No.  Prior to 1.9, this was an Alpha level feature, so it is not guaranteed to work during upgrades.  I will do some digging to get a better handle on exactly how it may fail.